### PR TITLE
Add unapply method for ThermalHouseResults

### LINKED ISSUE #934: Add unapply method for ThermalHouseResults
as part of issue #827

### PR DESCRIPTION
**Pull request by danielfeismann**
resolves #934 

required for #879

### CODE REVIEW

File: src/main/scala/edu/ie3/simona/event/ResultEvent.scala [hunk 1]
```diff
@@ -44,6 +53,50 @@ object ResultEvent {
...
 
+  object ThermalHouseResult {
+    def unapply(result: ThermalHouseResult): Option[
+      (
+          ZonedDateTime,
+          UUID,
+          ComparableQuantity[Power],
+          ComparableQuantity[Temperature],
+      )
+    ] = {
```
Comment (staudtMarius) [inline src/main/scala/edu/ie3/simona/event/ResultEvent.scala:64]: The brace can be removed.

File: src/main/scala/edu/ie3/simona/event/ResultEvent.scala [hunk 2]
```diff
@@ -44,6 +53,50 @@ object ResultEvent {
...
+
+  object CylindricalThermalStorageResult {
+    def unapply(result: CylindricalStorageResult): Option[
+      (
+          ZonedDateTime,
+          UUID,
+          ComparableQuantity[Power],
+          ComparableQuantity[Energy],
+      )
+    ] = {
```
Comment (staudtMarius) [inline src/main/scala/edu/ie3/simona/event/ResultEvent.scala:86]: See above.

**Review by staudtMarius — CHANGES_REQUESTED** [review]
Looks good to me. I found two parts that could be improved.

**Review by sebastian-peter — CHANGES_REQUESTED** [review]
I like this idea. Just some minor code formatting points.

**Review by sebastian-peter — APPROVED** [review]

**Pull request merged**